### PR TITLE
Add support for checking new connections

### PR DESCRIPTION
**Pull request by vojtech-suchy-sonarsource**

### CODE REVIEW

File: src/main/java/org/sonarsource/sonarlint/ls/SonarLintExtendedLanguageServer.java [hunk 1]
```diff
@@ -48,12 +48,44 @@ public interface SonarLintExtendedLanguageServer extends LanguageServer {
...
 
   class ConnectionCheckParams {
+    @Nullable
     private String connectionId;
 
+    @Nullable
+    private String token;
+
+    @Nullable
+    private String organization;
```
Comment (sophio-japharidze-sonarsource) [inline src/main/java/org/sonarsource/sonarlint/ls/SonarLintExtendedLanguageServer.java:58]: Sometimes we merge these two fields and call it `serverUrlOrOrganisationKey`

Comment (vojtech-suchy-sonarsource) [reply src/main/java/org/sonarsource/sonarlint/ls/SonarLintExtendedLanguageServer.java:58]: Firstly I had it implemented that way but didn't come up with a reliable way of detecting whether the `serverUrlOrOrganisationKey` actually belongs to SQ or SC (which I need to determine to construct the `ValidateConnectionParams`) - Do we already do that somewhere in language server?

**Review by sophio-japharidze-sonarsource — APPROVED** [review]
Looks like it does the job! I suggest not to merge it before client changes are also done and verified that it works

File: src/main/java/org/sonarsource/sonarlint/ls/SonarLintExtendedLanguageServer.java (hunk 1 above)
Comment (sophio-japharidze-sonarsource) [reply src/main/java/org/sonarsource/sonarlint/ls/SonarLintExtendedLanguageServer.java:58]: There is `isSonarCloudAlias(String serverUrl)` in `ServerConnectionSettings.java` but I'm not sure it is any help for your case

**Pull request merged**